cluster-autoscaler/cloudprovider/gce/gce_manager.go

```diff
@@ -88,51 +107,130 @@ func CreateGceManager(configReader io.Reader) (*GceManager, error) {
 		return nil, err
 	}
 	manager := &GceManager{
-		migs:      make([]*migInformation, 0),
-		service:   gceService,
-		migCache:  make(map[GceRef]*Mig),
-		zone:      zone,
-		projectId: projectId,
+		migs:        make([]*migInformation, 0),
+		gceService:  gceService,
+		migCache:    make(map[GceRef]*Mig),
+		zone:        zone,
+		projectId:   projectId,
+		clusterName: clusterName,
+		mode:        mode,
 		templates: &templateBuilder{
 			projectId: projectId,
 			zone:      zone,
 			service:   gceService,
 		},
 	}
+
+	if mode == ModeGKE {
+		gkeService, err := gke.New(client)
+		if err != nil {
+			return nil, err
+		}
+		manager.gkeService = gkeService
+		err = manager.fetchAllNodePools()
+		if err != nil {
+			glog.Errorf("Failed to fech node pools: %v", err)
+		}
+	}
+
 	go wait.Forever(func() {
 		manager.cacheMutex.Lock()
 		defer manager.cacheMutex.Unlock()
 		if err := manager.regenerateCache(); err != nil {
 			glog.Errorf("Error while regenerating Mig cache: %v", err)
 		}
 	}, time.Hour)
+
 	return manager, nil
 }
 
-// RegisterMig registers mig in Gce Manager.
-func (m *GceManager) RegisterMig(mig *Mig) {
-	m.cacheMutex.Lock()
-	defer m.cacheMutex.Unlock()
+func (m *GceManager) assertGKE() {
+	if m.mode != ModeGKE {
+		panic(fmt.Errorf("This should run only in GKE mode"))
+	}
+}
+
+// Gets all registered node pools
+func (m *GceManager) fetchAllNodePools() error {
+	m.assertGKE()
+
+	nodePoolsResponse, err := m.gkeService.Projects.Zones.Clusters.NodePools.List(m.projectId, m.zone, m.clusterName).Do()
+	if err != nil {
+		return err
+	}
+	for _, nodePool := range nodePoolsResponse.NodePools {
+		autoprovisioned := strings.Contains("name", nodeAutoprovisioningPrefix)
+		autoscaled := nodePool.Autoscaling != nil && nodePool.Autoscaling.Enabled
```

Comment:
Is autoprovisioned and autoscaled meant to be mutually exclusive? Or is the semantic different if it's both autoprovisioned and autoscaled?

Reply:
I mean I can figure out from code what will happen, my question is whether this is what we want to happen?

Reply:
Yes. Autoprovisioned doesn't have min/max.

Reply:
What if someone runs gcloud and updates autoscaling for autoprovisioned node pool?

Reply:
We won't allow that.